pi5/board.go

```diff
@@ -0,0 +1,364 @@
+//go:build linux
+
+// Package pi5 implements a raspberry pi5 board using pinctrl
+package pi5
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/viam-modules/pinctrl/pinctrl"
+	"go.uber.org/multierr"
+	pb "go.viam.com/api/component/board/v1"
+	"go.viam.com/rdk/components/board"
+	gl "go.viam.com/rdk/components/board/genericlinux"
+	"go.viam.com/rdk/grpc"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/utils"
+	rpiutils "raspberry-pi/utils"
+)
+
+// Model for rpi5.
+var Model = resource.NewModel("viam", "raspberry-pi", "rpi5")
+
+// pins are stored in /dev/gpiomem in order of gpio nums, so we must convert from pin name (physical num) to GPIO number.
+var pinNameToGPIONum = map[string]int{
```

Comment:
We will. I just wanted to get the model into this repo first since it was mostly copy/paste. Then I’m gonna update pwms in the other repo to use frequency & not be hardcoded, as well as start using data.go